src/type/form.ts

```diff
@@ -27,6 +27,13 @@ export interface UseFormParam<FormT> {
   lazy?: UseFormLazy
 }
 
+export interface ValidateOptions { fullResult?: boolean }
+export interface UseFormReturnRuleItem {
+  validate(v: any): boolean
+  validate(v: any, validateOptions?: ValidateOptions): { valid: boolean; message: string | null }
+}
```

Comment:
Maybe it could be written like this here (I don't know if it will work, just providing an idea)

```suggestion
export interface ValidateOptions<FullResult = boolean> { fullResult?: FullResult }
export interface UseFormReturnRuleItem {
  validate<FullResult extends Readonly<boolean> = false>(v: any, validateOptions?: ValidateOptions<FullResult>): FullResult extends true ? { valid: boolean; message: string | null } : boolean
}
```

Reply:
Using conditional type like this will surely bring us better type hint, though I find we can't do the same change to the [implementation code](https://github.com/xuzuodong/slimeform/blob/52b52ac5ab300a91d1264a6ef03fbfbe34f7edb3/src/rule.ts#L14), due to [this TypeScript limitation](https://stackoverflow.com/a/66553240/12931490) .

Reply:
Thank you for investigating this issue.
I looked at the limitation you mentioned and then reimplemented it with function overloading and it seems to work fine.
Check it out here: [TypeScript Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAShwFcBOA7AjFAvFARgezwBsIBDFAKAEsVgIkAzEgY2jkVQCYoBvcqKAG4lClACYAuXAWJk+UALYQAzkpIBzCJKXAk1NeQC+5KjTqMWUAPJhglPCiU859BIUJwlr4AH5J+IqQURia0DMzQ1rb2SgBiru7KXlAQAB60KKKOkXYOTvwubh5ekjoIEIbGAPSVUC4oTFEogsJiJLQAPHGFiYTAyWkQGY5wJKL2hCDtpRAAfFhQ0zMAFEKEZZJkIAA0UHg2OUqS2dGd8UW9MwCUkmzIKBzk1bUI9Y3NIqJtEKfdnr396UysFIYxQE3a-hkKDm2EYhCUsxWwnWUE2Oz2jSUvis+xOXQSf2AVxu8DuaHIdQaOXerVoSLWmlRKG2u1xDiObNiZx6wGuwPY9wpLyp9hpnzpqxRaNZmOxxwcfNuqHJlLeq1pEHpUuZ6M5cs5itJyqgAB9+XcuLx+EgjU0UPFUY5NhUqgAqKAAfS93p93qgABVlH0AMIkBGOX2Rr1QV2VcipMB4JB9VXUhMQBr+8Dff3LSWM-18-08KA2gXvMpQYJMaJ9Jhh5TzADacnTmez7SV6DzLXFmoARCQcExRP3Lpctq2UpB25BO7a0D2Pl8lmgOABmceTqcz4BZuddjhLjWrjc7bjPX7FBZISsGLc7jN7jtdxf0k9r9fny8E69whFVg+-Bts+B4Lsefanl+JaVO6ljAAAFnQMoHDGNT3hOxjAdOT77t8h4QSun7fgUv69JIACEFEYBhk7YbueHzgKb7qpBxElqR5zAJRFEAAyAZhAC6xjEH0dBIPMrFEWeHHcoSkgAOTaLoKBqApgEANyPDUUAAHreOQQA)

Reply:
Checkout my latest commit, now it's defined by function overloading. 
The only problem is that function overload should be declared in both [here](https://github.com/xuzuodong/slimeform/blob/7dbe06b24bf25556152d9b417ed6e70d2ae40644/src/type/form.ts#L34) and [here](https://github.com/xuzuodong/slimeform/blob/7dbe06b24bf25556152d9b417ed6e70d2ae40644/src/rule.ts#L14). I don't know if we can only declare this overload once.

Reply:
OK, then I will just leave it inside.